Fix SSLError in API integration tests

|Wazuh version|Component|
|---|---|
| 4.3.0 | Wazuh API Integration Tests |

<!--
Whenever possible, issues should be created for bug reporting and feature requests.
For questions related to the user experience, please refer:
- Wazuh mailing list: https://groups.google.com/forum/#!forum/wazuh
- Join Wazuh on Slack: https://wazuh.com/community/join-us-on-slack

Please fill the table above. Feel free to extend it at your convenience.
-->

Hello team,

We have a recurrent error in the API integration tests related to the SSL certificates when asking for a token:

```python
E   urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='localhost', port=55000): Max retries exceeded with url: /security/user/authenticate (Caused by SSLError(SSLError("bad handshake: SysCallError(-1, 'Unexpected EOF')")))
```

It also shows **invalid credentials** errors sometimes. Checking the `api.log` we can see it:

```shellsession
2021/07/08 16:08:14 INFO: HTTPS is enabled but cannot find the private key and/or certificate. Attempting to generate them
2021/07/08 16:08:15 INFO: Generated private key file in WAZUH_PATH/api/configuration/ssl/server.key
2021/07/08 16:08:15 INFO: Generated certificate file in WAZUH_PATH/api/configuration/ssl/server.crt
2021/07/08 16:10:53 INFO: Listening on 0.0.0.0:55000..
2021/07/08 16:10:53 INFO: unknown_user 172.20.0.1 "GET /security/user/authenticate" with parameters {} and body {} done in 0.038s: 401
2021/07/08 16:10:53 INFO: unknown_user 172.20.0.1 "GET /security/user/authenticate" with parameters {} and body {} done in 0.023s: 401
2021/07/08 16:10:53 INFO: unknown_user 172.20.0.1 "GET /security/user/authenticate" with parameters {} and body {} done in 0.011s: 401
2021/07/08 16:10:54 INFO: testing 172.20.0.1 "GET /security/user/authenticate" with parameters {} and body {} done in 0.430s: 200
2021/07/08 16:10:54 INFO: testing 172.20.0.1 "GET /syscollector/001/os" with parameters {"select": "os.name"} and body {} done in 0.182s: 200
2021/07/08 16:10:55 INFO: testing 172.20.0.1 "GET /security/user/authenticate" with parameters {} and body {} done in 0.490s: 200
```

We need to investigate if this error is environment related, implement a solution and also try to design a better mechanism to handle the token usage as we are asking for a new one for every test stage.

## Checks

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)
<!-- If changes are made to any of the following components, uncomment the corresponding line 
- [ ] **Integration tests** without failures for API configuration (`/wazuh-qa/tests/integration/test_api/test_config/`)
- [ ] **System tests** for agent enrollment process (`/wazuh-qa/tests/system/test_cluster/test_agent_enrollment`)
- [ ] **System tests** for agent info sync process in cluster (`/wazuh-qa/tests/system/test_cluster/test_agent_info_sync`)
- [ ] **System tests** for agent key polling (`/wazuh-qa/tests/system/test_cluster/test_agent_key_polling`)
- [ ] **System tests** for JWT invalidation (`/wazuh-qa/tests/system/test_jwt_invalidation`)
-->

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

Regards,
Víctor

--- a/api/test/integration/env/configurations/vulnerability/manager/healthcheck/healthcheck.py
+++ b/api/test/integration/env/configurations/vulnerability/manager/healthcheck/healthcheck.py
@@ -1,16 +1,17 @@
-import json
 import socket
 import sys
-from base64 import b64encode
 from os import system
 
 sys.path.append('/tools')
 
-from healthcheck_utils import get_manager_health_base, check, get_response, host, port, password, base_url, login_url, basic_auth, headers
+from healthcheck_utils import get_manager_health_base, get_login_header, check, get_response, base_url, login_url
 
 # Configuration
 user = 'wazuh'
+password = 'wazuh'
+headers = get_login_header(user, password)
 agents_to_check = ['001', '002']
+
 
 # Functions
 def get_agents_node(status='active'):

--- a/api/test/integration/env/configurations/vulnerability/manager/healthcheck/healthcheck.py
+++ b/api/test/integration/env/configurations/vulnerability/manager/healthcheck/healthcheck.py
@@ -6,52 +6,13 @@ from os import system
 
 sys.path.append('/tools')
 
-import requests
-import urllib3
-from healthcheck_utils import get_manager_health_base, check
-
-agents_to_check = ['001', '002']
+from healthcheck_utils import get_manager_health_base, check, get_response, host, port, password, base_url, login_url, basic_auth, headers
 
 # Configuration
-protocol = 'https'
-host = 'localhost'
-port = '55000'
 user = 'wazuh'
-password = 'wazuh'
-
-# Variables
-base_url = f"{protocol}://{host}:{port}"
-login_url = f"{base_url}/security/user/authenticate"
-basic_auth = f"{user}:{password}".encode()
-headers = {'Authorization': f'Basic {b64encode(basic_auth).decode()}'}
-
-
-# Disable insecure https warnings (for self-signed SSL certificates)
-urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
-
+agents_to_check = ['001', '002']
 
 # Functions
-def get_response(url, headers):
-    """Get API result for GET request.
-
-    Parameters
-    ----------
-    url : str
-        URL of the API (+ endpoint and parameters if needed).
-    headers : dict
-        Headers required by the API.
-
-    Returns
-    -------
-    Dict
-        API response for the request.
-    """
-    request_result = requests.get(url, headers=headers, verify=False)
-
-    if request_result.status_code == 200:
-        return json.loads(request_result.content.decode())
-
-
 def get_agents_node(status='active'):
     """Get to which manager each agent reports.
 

--- a/api/test/integration/env/tools/healthcheck_utils.py
+++ b/api/test/integration/env/tools/healthcheck_utils.py
@@ -3,9 +3,6 @@ import os
 import re
 import socket
 import time
-from base64 import b64encode
-import urllib3
-import requests
 
 # Configuration
 protocol = 'https'
@@ -15,13 +12,14 @@ user = 'testing'
 password = 'wazuh'
 
 # Variables
-base_url = f"{protocol}://{host}:{port}"
-login_url = f"{base_url}/security/user/authenticate"
-basic_auth = f"{user}:{password}".encode()
-headers = {'Authorization': f'Basic {b64encode(basic_auth).decode()}'}
+base_url = "{}://{}:{}".format(protocol, host, port)
+login_url = "{}/security/user/authenticate".format(base_url)
 
-# Disable insecure https warnings (for self-signed SSL certificates)
-urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+def get_login_header(user, password):
+    from base64 import b64encode
+    basic_auth = f"{user}:{password}".encode()
+    return {'Authorization': f'Basic {b64encode(basic_auth).decode()}'}
 
 
 def get_response(url, headers):
@@ -39,7 +37,11 @@ def get_response(url, headers):
     Dict
         API response for the request.
     """
-    request_result = requests.get(url, headers=headers, verify=False)
+    import urllib3
+    # Disable insecure https warnings (for self-signed SSL certificates)
+    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+    from requests import get
+    request_result = get(url, headers=headers, verify=False)
 
     if request_result.status_code == 200:
         return json.loads(request_result.content.decode())
@@ -90,14 +92,14 @@ def get_master_health():
     os.system("/var/ossec/bin/wazuh-control status > /tmp/daemons.txt")
     check0 = check(os.system("diff -q /tmp/output.txt /tmp/healthcheck/agent_control_check.txt"))
     check1 = check(os.system("diff -q /tmp/daemons.txt /tmp/healthcheck/daemons_check.txt"))
-    check2 = get_response(login_url, headers) is None
+    check2 = get_response(login_url, get_login_header(user, password)) is None
     return check0 or check1 or check2
 
 
 def get_worker_health():
     os.system("/var/ossec/bin/wazuh-control status > /tmp/daemons.txt")
     check0 = check(os.system("diff -q /tmp/daemons.txt /tmp/healthcheck/daemons_check.txt"))
-    check1 = check(os.system("grep -qs 'Listening on ' /var/ossec/logs/api.log"))
+    check1 = get_response(login_url, get_login_header(user, password)) is None
     return check0 or check1
 
 

--- a/api/test/integration/env/tools/healthcheck_utils.py
+++ b/api/test/integration/env/tools/healthcheck_utils.py
@@ -1,7 +1,49 @@
+import json
 import os
 import re
 import socket
 import time
+from base64 import b64encode
+import urllib3
+import requests
+
+# Configuration
+protocol = 'https'
+host = 'localhost'
+port = '55000'
+user = 'testing'
+password = 'wazuh'
+
+# Variables
+base_url = f"{protocol}://{host}:{port}"
+login_url = f"{base_url}/security/user/authenticate"
+basic_auth = f"{user}:{password}".encode()
+headers = {'Authorization': f'Basic {b64encode(basic_auth).decode()}'}
+
+# Disable insecure https warnings (for self-signed SSL certificates)
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+def get_response(url, headers):
+    """Get API result for GET request.
+
+    Parameters
+    ----------
+    url : str
+        URL of the API (+ endpoint and parameters if needed).
+    headers : dict
+        Headers required by the API.
+
+    Returns
+    -------
+    Dict
+        API response for the request.
+    """
+    request_result = requests.get(url, headers=headers, verify=False)
+
+    if request_result.status_code == 200:
+        return json.loads(request_result.content.decode())
+
 
 OSSEC_LOG_PATH = "/var/ossec/logs/ossec.log"
 
@@ -48,7 +90,7 @@ def get_master_health():
     os.system("/var/ossec/bin/wazuh-control status > /tmp/daemons.txt")
     check0 = check(os.system("diff -q /tmp/output.txt /tmp/healthcheck/agent_control_check.txt"))
     check1 = check(os.system("diff -q /tmp/daemons.txt /tmp/healthcheck/daemons_check.txt"))
-    check2 = check(os.system("grep -qs 'Listening on ' /var/ossec/logs/api.log"))
+    check2 = get_response(login_url, headers) is None
     return check0 or check1 or check2
 
 

